qiskit_experiments/library/randomized_benchmarking/mirror_rb_analysis.py

```diff
@@ -0,0 +1,674 @@
+            circ_result["metadata"]["polarization"] = pol
+            circ_result["metadata"]["polarization_uncertainty"] = pol_unc
+            if self.options.y_axis == "Effective Polarization":
+                y_data.append(pol)
+                y_data_unc.append(pol_unc)
+
+        return unp.uarray(y_data, y_data_unc)
+
+    def _initialize(
+        self,
+        experiment_data: ExperimentData,
+    ):
+        """Initialize curve analysis with experiment data.
+
+        This method is called ahead of other processing.
+
+        Args:
+            experiment_data: Experiment data to analyze.
+
+        Raises:
+            AnalysisError: When circuit metadata for ops count is missing.
+        """
+        super()._initialize(experiment_data)
+
+        if self.options.gate_error_ratio is not None:
+            # If gate error ratio is not False, EPG analysis is enabled.
+            # Here analysis prepares gate error ratio and gate counts for EPC to EPG conversion.
+
+            # If gate count dictionary is not set it will compute counts from circuit metadata.
+            avg_gpc = defaultdict(float)
+            n_circs = len(experiment_data.data())
+            for circ_result in experiment_data.data():
+                try:
+                    count_ops = circ_result["metadata"]["count_ops"]
+                except KeyError as ex:
+                    raise AnalysisError(
+                        "'count_ops' key is not found in the circuit metadata. "
+                        "This analysis cannot compute error per gates. "
+                        "Please disable this with 'gate_error_ratio=False'."
+                    ) from ex
+                nclif = circ_result["metadata"]["xval"]
+                for (qinds, gate), count in count_ops:
+                    formatted_key = tuple(sorted(qinds)), gate
+                    avg_gpc[formatted_key] += count / nclif / n_circs
+            self._gate_counts_per_clifford = dict(avg_gpc)
+
+            if self.options.gate_error_ratio == "default":
+                # Gate error dict is computed for gates appearing in counts dictionary
+                # Error ratio among gates is determined based on the predefined lookup table.
+                # This is not always accurate for every quantum backends.
+                gate_error_ratio = {}
+                for qinds, gate in self._gate_counts_per_clifford.keys():
+                    if set(qinds) != set(experiment_data.metadata["physical_qubits"]):
+                        continue
+                    gate_error_ratio[gate] = _lookup_epg_ratio(gate, len(qinds))
+                self.set_options(gate_error_ratio=gate_error_ratio)
+
+        # Get qubit number
+        self._physical_qubits = experiment_data.metadata["physical_qubits"]
+        self._num_qubits = len(experiment_data.metadata["physical_qubits"])
+
+
+def _lookup_epg_ratio(gate: str, n_qubits: int) -> Union[None, int]:
```

Comment:
Same. You can move them to some util file to avoid having same function in multiple places.

Reply:
Thanks, no need for these now that this is a subclass.